ios/Exponent/Versioned/Core/Api/EXScreenOrientation.m

```diff
@@ -33,37 +41,210 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId
   return self;
 }
 
-RCT_EXPORT_METHOD(allowAsync:(NSString *)orientation
+RCT_EXPORT_METHOD(lockAsync:(EXOrientationLock)orientationLock
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject)
 {
-  UIInterfaceOrientationMask orientationMask = [self orientationMaskFromOrientation:orientation];
+  UIInterfaceOrientationMask orientationMask = [self orientationLockJSToNative:orientationLock];
   if (orientationMask == INVALID_MASK) {
-    return reject(@"E_INVALID_ORIENTATION", [NSString stringWithFormat:@"Invalid screen orientation %@", orientation], nil);
+    return reject(@"ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK", [NSString stringWithFormat:@"Invalid screen orientation lock %@", [self orientationLockToString:orientationLock]], nil);
   }
   if (![self doesSupportOrientationMask:orientationMask]) {
-    return reject(@"E_UNSUPPORTED_ORIENTATION", [NSString stringWithFormat:@"This device does not support this orientation %@", orientation], nil);
+    return reject(@"ERR_SCREEN_ORIENTATION_UNSUPPORTED_ORIENTATION_LOCK", [NSString stringWithFormat:@"Cannot apply orientation lock %@ because this device does not support the PORTRAIT_DOWN orientation", [self orientationLockToString:orientationLock]], nil);
   }
   [_kernelOrientationServiceDelegate screenOrientationModule:self
                      didChangeSupportedInterfaceOrientations:orientationMask];
   resolve(nil);
 }
 
-RCT_EXPORT_METHOD(doesSupportAsync:(NSString *)orientation
+
+RCT_EXPORT_METHOD(lockPlatformAsync:(NSArray *)allowedOrientations
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject)
 {
-  UIInterfaceOrientationMask orientationMask = [self orientationMaskFromOrientation:orientation];
-  if (orientationMask == INVALID_MASK) {
-    return reject(@"E_INVALID_ORIENTATION", [NSString stringWithFormat:@"Invalid screen orientation %@", orientation], nil);
+  // combine all the allowedOrientations into one bitmask
+  UIInterfaceOrientationMask allowedOrientationsMask = 0;
+  for (NSString * allowedOrientation in allowedOrientations ){
```

Comment:
`for (NSString *allowedOrientation in allowedOrientations) {` (fix spacing)
in general, the standard for obj c seems to be no space between the `*` and variable name, so could you fix all the cases in this file where you have an extra space?